<a href="https://colab.research.google.com/github/IrinaIsobel/project2/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import pandas as pd

##Data Set No. 1

In [4]:
metabolic = pd.read_csv('https://query.data.world/s/bcgzionzc7ivfiv7jgf5vbdmvklfgc?dws=00000')
metabolic.head()

seqn  Age     Sex  Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male   Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female  Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male   Single   800.0  Asian       69.6  20.1            0   
3  62172   43  Female   Single  2000.0  Black      120.4  33.3            0   
4  62177   51    Male  Married     NaN  Asian       81.1  20.1            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn  
3     5.22       5.0           104   73            141         No MetSyn  
4     8.13       5.0            95   43            126         No MetSyn

In [8]:
metabolic.shape

(2401, 15)

1. Source of data
- [Metabolic Syndrome Prediction Dataset](https://data.world/informatics-edu/metabolic-syndrome-prediction.). Retrieved 03/21/23.

2. Brief description of data
- This dataset is used to predict whether a patient is likely to get Metabolic Syndrome based on the input parameters like age, sex, race, and BMI. Each row in the data provides relavant information about the patient.

3. What is the target?
- The target is MetabolicSyndrome (yes or no)

4. What does one row represent? (A person?  A business?  An event? A product?)
- Each row represents a person and their personal medical data.

5. Is this a classification or regression problem?
- This is a classification problem.

6. How many features does the data have?
- The data has 14 features.

7. How many rows are in the dataset?
- The data set has 2,401 rows.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
- The challenges I see in this data set is choosing which specific features will be relevant to the target, and how to impute any missing data.

##Data Set No. 2

In [3]:
stroke = pd.read_csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vQwT4Q_Hiezgbu98UOI6Qb9BbkmnK8ghZ7-xowBNpr2_h2CHcE9JJNC4vzyAs4rxdRlIdmYCFBoldlj/pub?output=csv')
stroke.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [9]:
stroke.shape

(5110, 12)

1. Source of data
- [Stroke Prediction Dataset](https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset?resource=download). Retrieved 03/21/23.

2. Brief description of data
- This dataset is used to predict whether a patient is likely to have a stroke based on the input parameters like gender, age, hypertension, and heart diease. Each row in the data provides relavant information about the patient.

3. What is the target?
- The target is stroke (0 or 1).

4. What does one row represent? (A person?  A business?  An event? A product?)
- One row represents a person and their personal and medical data.

5. Is this a classification or regression problem?
- This is a classification problem.

6. How many features does the data have?
The data has 11 features.

7. How many rows are in the dataset?
- There are 5,110 rows in the data set.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

- One challenge I forsee is not having enough pertinent information about a person to predict a stroke. I see a challenges with balancing prescion and recall.